src/app/static/src/tests/adrUpload/mutations.test.ts

```diff
@@ -56,4 +58,33 @@ describe("ADR mutations", () => {
         expect(state.currentFileUploading).toBe(null);
         expect(state.totalFilesUploading).toBe(null);
     });
+
+    it("can clear status", () => {
+        const state = mockADRUploadState();
+        mutations[ADRUploadMutation.ClearStatus](state);
+        expect(state.uploadComplete).toBe(false);
+        expect(state.releaseCreated).toBe(false);
+        expect(state.releaseFailed).toBe(false);
+    });
+
+    it("can set release created", () => {
+        const state = mockADRUploadState();
+        mutations[ADRUploadMutation.ReleaseCreated](state);
+        expect(state.releaseCreated).toBe(true);
+    });
+
+    it("can set release failed with error", () => {
+        const state = mockADRUploadState();
+        mutations[ADRUploadMutation.ReleaseFailed](state, {payload: mockError("Version already exists for this activity")});
+        expect(state.releaseFailed).toBe(true);
+        expect(state.uploadError!!.detail).toBe("A release already exists on ADR for the latest files");
```

Comment:
I'm not sure how I would test the different translations here.

Reply:
I'm not greatly concerned by this, as long as you can modify `downloadResults.test.ts` to check for the translated error detail message (at the moment it only checks for the summary?).

But you should really do a test here - at least for one other language. I'm a bit hazy on this but I think the easiest might be to simply do `i18next.changeLanguage("fr");` before making the mutation.

Reply:
This isn't really resolved - we don't have any checks that the translations for "Release names must be unique per dataset..." and "A release already exists..." are actually displayed. As stated above, you could do this this here (by changing the language before invoking the mutation) and/or by checking the rendered messages in `downloadResults.test.ts`.

Reply:
I was able to check the French translations but when it came round to the Portuguese, using `i18next.changeLanguage("pt");` still returned English. You mentioned testing one language would be enough though.

Reply:
You need to register `pt` [here](https://github.com/mrc-ide/hint/blob/master/src/app/static/src/tests/setup.ts#L23)